sdk/openai/Azure.AI.OpenAI/src/Generated/AudioTranscriptionOptions.Serialization.cs

```diff
@@ -193,6 +201,59 @@ internal static AudioTranscriptionOptions DeserializeAudioTranscriptionOptions(J
                 serializedAdditionalRawData);
         }
 
+        private BinaryData SerializeMultipart(ModelReaderWriterOptions options)
+        {
+            using MultipartFormDataRequestContent content = ToMultipartRequestContent();
+            using MemoryStream stream = new MemoryStream();
+            content.WriteTo(stream);
+            if (stream.Position > int.MaxValue)
+            {
+                return BinaryData.FromStream(stream);
+            }
+            else
+            {
+                return new BinaryData(stream.GetBuffer().AsMemory(0, (int)stream.Position));
+            }
+        }
+
+        internal virtual MultipartFormDataRequestContent ToMultipartRequestContent()
+        {
+            MultipartFormDataRequestContent content = new MultipartFormDataRequestContent();
+            content.Add(AudioData, "file", "file");
+            if (Optional.IsDefined(Filename))
```

Comment:
Why do we need to use Optional here instead of checking for null?

Reply:
`FileName` is an optional property, we need to check if is defined, if so we will serialize it. Just as we serialize to json.

Reply:
this is our current generated pattern that the optionality of an optional property is always checked in this way.
We made this decision because we have to do a long check if the property is a collection, and we would like to wrap that check as a method, to align the style, we just do the same for simple properties as well.